src/nodes/functions/material/getShIrradianceAt.js

```diff
@@ -0,0 +1,28 @@
+import { Fn, mul } from '../../tsl/TSLBase.js';
+
+const getShIrradianceAt = /*@__PURE__*/ Fn( ( [ normal, shCoefficients ] ) => {
+
+	// normal is assumed to have unit length
+
+	const x = normal.x, y = normal.y, z = normal.z;
+
+	// band 0
+	const result = shCoefficients.element( 0 ).mul( 0.886227 );
+
+	// band 1
+	result.addAssign( shCoefficients.element( 1 ).mul( 2.0 * 0.511664 ).mul( y ) );
```

Comment:
I believe I recall @sunag preferring this pattern now...
```js
result = result.add( ... )
```

requires `let`. Whatever he prefers is fine with me.

Reply:
`.addAssign()` relies on a global stack, while `result = result.add()` preserves the process tree. We may see many times when compiling everything inline using `let` instead of `assign`, this is because the compiler did not find it necessary to do apply `assigns` to builded code. `assign()` can also only be executed inside a `Fn` that has a controlled environment for stacks. Using `let` is a little better for the compiler to understand what it should and should not optimize in this sense, but it is not at all strict, if you need to use `addAssign()` that is fine.

Reply:
I'll change the code to `result = result.add()`.